src/docs/theia_ai.md

```diff
@@ -0,0 +1,404 @@
+
+*Agent-specific Variables* are only available to specific agents (and in their prompt templates), but not to other agents and also not in the default chat.
+
+#### Agent-specific Variables
+
+Agent-specific variables are controlled by a specific agent. To fill a variable with data, you can literally use any internal Theia API or other external APIs or methods you like. As an example, let’s again look at the chat agent that assists users in identifying (and later executing) arbitrary commands in the Theia IDE, such as opening the settings or showing the toolbar. You can review the [full code of this agent](https://github.com/eclipse-theia/theia/blob/master/packages/ai-chat/src/common/command-chat-agents.ts) as a reference.
+
+To allow the underlying LLM to identify the right command, we want to send it a list with all available commands. To achieve this, we first retrieve a list of all available commands using the CommandRegistry (see code example below). This step is independent of Theia AI and you could retrieve any sort of data here instead.
+Second, we resolve the variable ‘command-ids’ when receiving the system prompt’ (also see code example below).
+
+```ts
+// Retrieve all available commands
+const knownCommands: string[] = [];
+for (const command of this.commandRegistry.getAllCommands()) {
+   knownCommands.push(`${command.id}: ${command.label}`);
+}
+
+// Resolve the variable when retrieving the system prompt
+const systemPrompt = await this.promptService.getPrompt('command-chat-agent-system-prompt-template', {
+   'command-ids': knownCommands.join('\n')
+});
+```
+
+Please note that the manual resolvement of variables is not necessary when using global variables.
+Now we can simply use our agent-specific variable in our prompt templates, e.g. like this:
+
+```md
+You are a service that helps users find commands to execute in an IDE.
+
+
+These are the available commands:
+Begin List:
+{{command-ids}}
+End List
+
+
+```
+
+If an agent uses agent-specific variables, it can decide to explicitly declare them. This enables one to keep track of used variables and optionally even show this information to the user (see example screenshot below). 
+
+<img src="../../used-variables-command-agent.png" alt="Used variables in Theia AI" style="max-width: 525px">
+
+To declare an agent-specific variable, simply add it like this in the constructor of your agents, along with an optional description and whether it is used in the prompt template or not.
+
+
+```ts
+this.agentSpecificVariables = [{
+           name: 'command-ids',
+           description: 'The list of available commands in Theia.',
+           usedInPrompt: true
+       }];
+```
```

Comment:
```suggestion
```ts
this.agentSpecificVariables = [{
   name: 'command-ids',
   description: 'The list of available commands in Theia.',
   usedInPrompt: true
}];
```
```